### PR TITLE
update avro to 1.11.3

### PR DESCRIPTION
**Pull request by raphaelauv**

### CODE REVIEW

**Comment by mavogel** [comment]
@davidmc24 could we push this PR forward? :) let's run the workflows

**Comment by davidmc24** [comment]
Sorry I didn't see this sooner.  Taking a look now.

File: README.md [hunk 1]
```diff
@@ -26,8 +26,8 @@ This is a [Gradle](http://www.gradle.org/) plugin to allow easily performing Jav
     * Java 8-12 support requires Gradle 5.1 or higher (versions lower than 5.1 are no longer supported)
 * Currently built against Gradle 7.6
     * Currently tested against Gradle 5.1-5.6.4 and 6.0-7.6
-* Currently built against Avro 1.11.2
-    * Currently tested against Avro 1.11.0-1.11.2
+* Currently built against Avro 1.11.3
+    * Currently tested against Avro 1.11.0-1.11.2-1.11.3
```
Comment (davidmc24) [inline README.md:30]:
```suggestion
    * Currently tested against Avro 1.11.0-1.11.3
```

**Comment by davidmc24** [comment]
Looks like `AvroUtilsSpec.assemblePath rejects unnamed arguments` fails with a NullPointerException.

https://scans.gradle.com/s/cmglhacj5ozuq/tests/task/:test/details/com.github.davidmc24.gradle.plugin.avro.AvroUtilsSpec/assemblePath%20rejects%20unnamed%20arguments%20(%23arg)?expanded-stacktrace=WyIwIl0&top-execution=1

Fix that, if you wouldn't mind.  Otherwise it looks pretty good.

**Comment by mavogel** [comment]
@raphaelauv let me know if you need support on this. We're also interested in the merge / release of the new version :)

**Comment by raphaelauv** [comment]
@mavogel yes you can continue this PR , I'm not available to continue, thanks for your help

**Comment by mavogel** [comment]
After some investigation, the NullPointerException happens as in
- avro `v1.11.3`: they changed the logic of the contructor:  https://github.com/apache/avro/blob/release-1.11.3/lang/java/avro/src/main/java/org/apache/avro/Protocol.java#L286 and try to parse the `name` parameter without a `null` check. There are also no tests for this case: https://github.com/apache/avro/blob/release-1.11.3/lang/java/avro/src/test/java/org/apache/avro/TestProtocol.java#L28
- whereas in avro `v1.11.2`: you could pass `null` as `name` and it was simply assigned to the member variable and it was checked in this plugin code here: https://github.com/davidmc24/gradle-avro-plugin/blob/1.8.0/src/main/java/com/github/davidmc24/gradle/plugin/avro/AvroUtils.java#L62

I am not 100% sure if it is a bug in `apache/avro` or in this plugin. My question: can `name` be empty according to the spec? WDYT @davidmc24 ?

**Comment by davidmc24** [comment]
@mavogel Thanks for the investigation, Manuel.

Further analysis:

* Both in the old and new version of Avro, if parsing from a file, `parseNameAndNamespace` throws a `SchemaParseException` if the Protocol lacks a name (as specified by the `protocol` field)
* In the old version, if you construct a Protocol with a null name, it completes construction, but would fail with an unchecked NPE on equals or hashCode.
* In the new version, if you construct a Protocol with a null name, it fails construction with an NPE in parsing logic, as you said.
* The specification appears to say that (at least when using a file), the protocol (name) attribute is required; https://avro.apache.org/docs/1.11.1/specification/#protocol-declaration

My conclusion is that while perhaps not a "bug" in the new Avro version (as it's a behavior that only manifests when invalid arguments are used), it's still sloppy coding practice.  From the perspective of this plugin, ideally I'd like it to "work correctly" (in this case, meaning "give a useful error message") when either version is used.

That said... it isn't really the end-user experience which is changing here, but that a unit test for the internal logic of `AvroUtils` is no longer testable.  That is, the use-case of handling a constructed `Protocol` object with a null name still "works" in the plugin, it's just not possible to get into that state in the new version barring extraordinary measures, such as reflection.

Thus, my proposal is that you remove that test case from the test.  Specifically, the `createProtocol(null, null)             | _` line from `AvroUtilsSpec`.

**Comment by mavogel** [comment]
Thanks a lot for your additions and sounds good to me @davidmc24 . This PR can then be closed in favor of #239 then (as  I needed to create my own fork due to permissions)

**Comment by raphaelauv** [comment]
-> https://github.com/davidmc24/gradle-avro-plugin/pull/239

**Pull request closed without merging**